### PR TITLE
BBB-408 Client credentials - have different client id and secret for …

### PR DESCRIPTION
**Pull request by stephen-bealine**
…each BBB service. Introduce test third party app to test client creds

### CODE REVIEW

File: auth-service/build.gradle [hunk 1]
```diff
@@ -49,6 +49,12 @@ sourceSets {
 	}
 }
 
+idea{
```
Comment (mgilgar) [inline auth-service/build.gradle]: What is this for?

File: auth-service/src/main/java/uk/gov/dft/bluebadge/authorisationservice/security/BBClientDetailsService.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        AuthorityUtils.createAuthorityList("bb_resource_server");
+    configuredClients =
+        clients
+            .stream()
+            .peek(c -> c.setAuthorities(auths))
+            .collect(Collectors.toMap(ClientDetails::getClientId, Function.identity()));
+  }
+
+  @Override
+  public ClientDetails loadClientByClientId(String clientId) throws ClientRegistrationException {
```
Comment (mgilgar) [inline auth-service/src/main/java/uk/gov/dft/bluebadge/authorisationservice/security/BBClientDetailsService.java]: I think we decide to use the word "retrieve".

Comment (stephen-bealine) [reply auth-service/src/main/java/uk/gov/dft/bluebadge/authorisationservice/security/BBClientDetailsService.java]: This is from the interface, so no option of rename

File: auth-service/build.gradle (hunk 1 above)
Comment (stephen-bealine) [reply auth-service/build.gradle]: To tell Intellij that the acceptance test source folder is a test folder and not main code

**Pull request merged**